### PR TITLE
cmd/trace-agent: remove remote rates and config endpoint APM feature flag

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Enables remote rates & config endpoint by default when remote-config is enabled.

### Motivation

Open the door for APM/LD RC beta customers.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by mellon85 — APPROVED** [review]

**Review by coignetp — APPROVED** [review]

File: cmd/trace-agent/config/config.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func prepareConfig(path string) (*config.AgentConfig, error) {
 		return cfg, err
 	}
 	cfg.ConfigPath = path
-	if features.Has("remote_rates") {
+	if coreconfig.Datadog.GetBool("remote_configuration.enabled") {
```
Comment (gbbr) [inline cmd/trace-agent/config/config.go:80]: Don't you need to set these as known? Also, most stuff is suffixed `_config`, why are you suffixing this `_configuration`?

Comment (arbll) [reply cmd/trace-agent/config/config.go:80]: This is an existing flag that is shared with the core-agent, I can't change the name unfortunately. Do I still need to `SetKnown` in that context ?

Comment (gbbr) [reply cmd/trace-agent/config/config.go:80]: In that case we're good, `*SetDefault` calls `SetKnown` under the hood and I see it's been added here: https://github.com/DataDog/datadog-agent/blob/ba01c560a72151903ab6c1274dc49f809e3a0462/pkg/config/config.go#L284

**Review by raphaelgavache — APPROVED** [review]

**Review by olivielpeau — APPROVED** [review]
(approving for agent-core-owned file, `pkg/config/config.go`)

**Pull request merged**